### DISCUSSION TITLE
Source: Bold Subscriptions

### DISCUSSION THREAD

**Discussion by beam-ryan** [airbytehq/airbyte#35256] [New Connector Request]
https://developer.boldcommerce.com/default/documentation/subscriptions

[Customers](https://developer.boldcommerce.com/default/documentation/subscriptions#/Customers/getCustomers)
[Subscriptions](https://developer.boldcommerce.com/default/documentation/subscriptions#/Subscriptions/getSubscriptions)

## Tell us about the new integration you’d like to have
Which source and which destination? Which frequency?

Looking to extract [Bold Subscription](https://developer.boldcommerce.com/default/documentation/subscriptions
) data to our snowflake data warehouse. Looking to extract nightly. 

## Describe the context around this new integration
Which team in your company wants this integration, what for? This helps us understand the use case.

This will be used to track key metrics: 

* how many new subscribers
* what is our subscription churn
* what products have the highest subscription retention
* what marketing channels drive higher subscription conversion

## Describe the alternative you are considering or using
What are you considering doing if you don’t have this integration through Airbyte?

So far I cannot find any other ETL tool that supports bold subscription data. I see airbyte has Recurly, this integration would be very similar. Bold is a shopify plugin and we are having trouble making custom reports.

Please help! thanks!

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368152477925) by [Unito](https://www.unito.io)

**Comment by sherifnada** [comment]
Posting updates I mentioned to @beam-ryan [in Slack](https://airbytehq.slack.com/archives/C01AHCDHCKE/p1615187184041700?thread_ts=1615090335.037100&cid=C01AHCDHCKE) to have the context in one place: 


> That article is a good one to get a solid idea of the basics, but it’s intentionally extremely barebones for educational purposes. You’ll may also want to read the ‘sequel’ which describes how to add incremental sync to a source: https://docs.airbyte.io/tutorials/adding-incremental-sync
> To see a more “prod ready” example of building a python source, see this article: https://docs.airbyte.io/tutorials/building-a-python-source
> there’s a few examples of connectors we’re currently running in production. The closest ones to bold (in terms of of API functionality) is probably sendgrid: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-sendgrid or even simpler (relies on fewer helper libraries which may make it easier to understand than sendgrid), mailchimp: https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-mailchimp/source_mailchimp
> we could start dev work on Bold in ~3-4 weeks, which means it’d land in master in ~5-6 weeks. One risk to this estimate is getting a sandbox account to verify the account works. we’ll need to connect a Bold shop to the API etc. so depending on how straight forward that process is, the timeline may be impacted

**Comment by sherifnada** [comment]
instructions from bold ecommerce on getting a sandbox account

```
1. Create a development store on one of the available platforms in WooCommerce, CommerceTools or BigCommerce and add the Bold Checkout Application. Please note for CommerceTools and WooCommerce there are extra steps needed on the Bold end to get this enabled for you.
2. Once the development store is available and Bold Checkout is installed, please fill out the form attached here: https://start.boldcommerce.com/cashier-plugin-request-information.
3. Once this is completed, we will create the plugin and whitelist it on our end. This will allow the plugin to be available within the Marketplace section of the checkout area only on the specified development store. This is where it will live until there is a plugin created correctly and effectively. If you need multiple stores with this plugin, we can make that work!
4. An example plugin to get started can be found here: https://github.com/bold-commerce/example-node-cashier-plugin
```

@arhip11 can we prioritize this for next sprint?

**Comment by beam-ryan** [comment]
This would be amazing!!! Thank you so much.

[a comment by an automated account is not shown]
